lexers/ansi_c.lua

```diff
@@ -51,6 +51,11 @@ local include = lex:tag(lexer.PREPROCESSOR, '#' * S('\t ')^0 * 'include') *
 local preproc = lex:tag(lexer.PREPROCESSOR, '#' * S('\t ')^0 * lex:word_match(lexer.PREPROCESSOR))
 lex:add_rule('preprocessor', include + preproc)
 
+-- Attributes.
+lex:add_rule('attribute', lex:tag(lexer.ATTRIBUTE, '[[' * 
+  (lex:word_match(lexer.ATTRIBUTE) + (lexer.word * ':' * lexer.word)) * 
+  lexer.range('(', ')')^-1 * ']]'))
```

Comment:
Do you think there is any value in having annotation arguments highlighted normally? For example, should `[[deprecated("reason")]]` highlight `"reason"` as a string? (And perhaps highlight the parentheses as operators?)

It may be worth trying to highlight arguments separately. If so, we may want to consider highlighting `[[` and `]]` as single entities (either as attributes or operators) and then use `B('[[') * lex:tag(lexer.ATTRIBUTE, lex:word_match(lexer.ATTRIBUTE) + lexer.word * '::' * lexer.word)` to match attributes.

This would not work for attribute lists (e.g. `[[foo, bar, baz]]` would highlight foo as an attribute, but not bar or baz), but it would catch a vast majority of cases.

Reply:
I'll defer here. In my use case I'd rather have the entire attribute highlit as one thing. Having the string marked as such makes sense--but imo having the parens marked as well would be too much visual clutter, and doing one without the other would seem arbitrary.

Attribute lists are an important consideration though, I'll add a patch for that momentarily.